File: src/Sample.Invoker/Models/Point.cs
using System.Runtime.InteropServices;

namespace Sample.Invoker.Models
{
    [StructLayout(LayoutKind.Sequential)]
    public readonly struct Point
    {
        public Point(string name, int x, int y, string[] titles, float[] angles)
        {
            this.Name = name;
            this.X = x;
            this.Y = y;
            this.Titles = titles;
            this.Angles = angles;
        }

        public readonly string Name;

        public readonly int X;

        public readonly int Y;

        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 2)]
        public readonly string[] Titles;

        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 2)]
        public readonly float[] Angles;
    }
}

File: src/Sample.NativeLibrary/point.cpp
#include <iostream>
#include "point.h"
using namespace std;

Point::Point() = default;

Point::Point(const char* name, const int x, const int y)
	:Point()
{
	this->Name = name;
	this->X = x;
	this->Y = y;
}

Point::~Point()
{
	std::cout << "C++ Point析构函数调用" << endl;
}
